tensorflow/python/kernel_tests/gather_nd_op_test.py

```diff
@@ -207,7 +219,20 @@ def testBadIndices(self):
           r"\(shape: \[3\]\)"):
         gather_nd.eval()
 
-  def testBadIndicesWithSlices(self):
+  def testBadIndicesWithSlicesCPU(self):
+    with self.test_session(use_gpu=False):
+      params = [[0, 1, 2]]
+      indices = [[[0], [0], [1]]]  # Make this one higher rank
+      gather_nd = array_ops.gather_nd(params, indices)
+      with self.assertRaisesOpError(
+          r"flat indices\[2, :\] = \[1\] does not index into param "
+          r"\(shape: \[1,3\]\)"):
+        gather_nd.eval()
+
+  def _disabledTestBadIndicesWithSlicesGPU(self):
```

Comment:
check if a gpu is available, and if not then return.

Reply:
(re-enable the test)

Reply:
@dantkz How would we address the TODO?

Reply:
@Androbin Do you mean "TODO: check if SYCL supports these"? It's fine to remove the changes and remove the TODO comment.

Reply:
No, I mean these:
```
TODO disabled due to different behavior of gather_nd and gather on GPU and CPU
On GPU the bad indices do not raise error but fetch 0 values
```

Reply:
Oh, I don't think there is anything that can be done. That test function is there in case the default behaviour of GPU gather changes.